wfe2/wfe.go

```diff
@@ -2031,6 +2031,10 @@ func (wfe *WebFrontEndImpl) NewOrder(
 		Names:          names,
 	})
 	if err != nil || order == nil || order.Id == 0 || order.Created == 0 || order.RegistrationID == 0 || order.Expires == 0 || len(order.Names) == 0 {
+		var bErr *berrors.BoulderError
+		if errors.As(err, &bErr) && bErr.Type == berrors.RateLimit && bErr.RetryAfterSeconds() > 0 {
+			response.Header().Add("Retry-After", strconv.Itoa(bErr.RetryAfterSeconds()))
+		}
```

Comment:
I think this logic would have a nice home in `wfe.sendError`. That way it's fairly generic: if a component returns a BoulderError with a non-zero RetryAfter, we will serve a Retry-After header. And I think it makes sense to do this behavior for all error types. For instance, one can imagine InternalServer and OrderNotReady errors having Retry-After.

Reply:
Moving the logic to `wfe.sendError()` also made is super easy for me to add some tests around it. Thanks!